integrations/google-mail/syncs/emails.ts

```diff
@@ -55,21 +56,40 @@ export default async function fetchData(nango: NangoSync) {
 function mapEmail(messageDetail: Schema$Message, headers: Record<string, any>): GmailEmail {
     const parts = messageDetail.payload?.parts || [];
     let body = '';
-    for (const part of parts) {
-        if (part.mimeType === 'text/plain' && part.body?.data) {
-            // Body can be empty if the part is an attachment
-            // https://developers.google.com/gmail/api/reference/rest/v1/users.messages.attachments#MessagePartBody
-            body = Buffer.from(part.body.data, 'base64').toString('utf8');
-            break;
+    const attachments: Attachments[] = [];
+
+    function processParts(parts: Schema$MessagePart[]) {
```

Comment:
for clarity it could be a put outside the fetchData function

Reply:
Outside the `mapEmail` function 😅, updated with the latest [commit](https://github.com/NangoHQ/integration-templates/pull/41/commits/9bfe3316b5274c884f4596a207a7bf5949c6e90f).